src/main/java/com/idrsolutions/microservice/JPedalServlet.java

```diff
@@ -270,88 +273,99 @@ protected boolean validateRequest(final HttpServletRequest request, final HttpSe
-                                paramMap.get("password"),
-                                outputDirStr + fileSeparator,
-                                -1,
-                                durationTracker);
+                        commandArgs.add("org.jpedal.examples.text.ExtractTextAsWordlist");
+                        commandArgs.add(inputPdf.getAbsolutePath());
+                        commandArgs.add(outputDir.getAbsolutePath());
                         break;
                     case "structuredText" :
-                        ExtractStructuredText checkForStructure = new ExtractStructuredText(userPdfFilePath);
-                        if (checkForStructure.openPDFFile()) {
-                            Document content = checkForStructure.getStructuredTextContent();
-                            if (content != null && content.hasChildNodes() && content.getDocumentElement().hasChildNodes()) {
-                                ExtractStructuredText.writeAllStructuredTextOutlinesToDir(
-                                        userPdfFilePath,
-                                        paramMap.get("password"),
-                                        outputDirStr + fileSeparator + fileNameWithoutExt + fileSeparator,
-                                        durationTracker,
-                                        null);
-                            } else {
-                                throw new JPedalServletException("File contains no structured content to extract.");
-                            }
-                        } else {
-                            throw new JPedalServletException("Unable to open specified file");
-                        }
+                        commandArgs.add("org.jpedal.examples.text.ExtractStructuredText");
+                        commandArgs.add(inputPdf.getAbsolutePath());
+                        commandArgs.add(outputDir.getAbsolutePath());
                         break;
                 }
                 break;
-            default:
-                throw new JPedalServletException("Unrecognised mode specified: " + mode.name());
-        }
-    }
+                default:
+                    try {
+                        throw new JPedalServletException("Unrecognised mode specified: " + mode.name());
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                }
 
-    static class JPedalServletException extends Exception {
-        JPedalServletException(final String msg) {
-            super(msg);
-        }
+
+                final String[] commands = commandArgs.toArray(new String[0]);
+
+                return ProcessUtils.runProcess(commands, inputPdf.getParentFile(), uuid, "JPedal Conversion", maxDuration);
     }
 }
+
+        class JPedalServletException extends Exception {
+            JPedalServletException(final String msg) {
+                    super(msg);
+                }
+        }
```

Comment:
Depending on if / how the try/catch under default is handled, do we still need to keep this class definition?